### PR TITLE
OSDOCS-9501# Add clone support for Azure File (TP)

### PR DESCRIPTION
**Pull request by lpettyjo**
Version(s): 4.16

Issue: https://issues.redhat.com/browse/OSDOCS-9501

Link to docs preview: https://73733--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi#csi-drivers-supported_persistent-storage-csi

QE review:
- [x] QE has approved this change.

Additional information:

**PTAL**: @RomanBednar @ropatil010 @gcharot

### CODE REVIEW

File: modules/persistent-storage-csi-drivers-supported.adoc [hunk 1]
```diff
@@ -24,6 +24,16 @@ supported by {product-title}
...
 and which CSI features they support, such as volume snapshots and resize.
 
+ifndef::openshift-rosa[]
+[IMPORTANT]
+====
+If your CSI driver is not listed in the preceding table, you must follow the installation instructions provided by your CSI storage vendor to use their supported CSI features.
+====
+endif::openshift-rosa[]
+ifdef::openshift-rosa[]
+In addition to the drivers listed in the preceding table, ROSA functions with CSI drivers from third-party storage vendors. Red Hat does not oversee third-party provisioners or the connected CSI drivers and the vendors fully control source code, deployment, operation, and Kubernetes compatibility. These volume provisioners are considered customer-managed and the respective vendors are responsible for providing support. See the link:https://docs.openshift.com/rosa/rosa_architecture/rosa_policy_service_definition/rosa-policy-responsibility-matrix.html#rosa-policy-responsibilities_rosa-policy-responsibility-matrix[Shared responsibilities for {product-title}] matrix for more information.
```
[the remark this answers is not in the document]
Comment (mramendi) [reply modules/persistent-storage-csi-drivers-supported.adoc:34]: I now see the paragraphs were actually moved from "after the table" to "before the tablwe" in this PR - is this a technical error?

**Review by mramendi — COMMENTED** [review]
That's it for peer review from me - the position of the paragraph is kinda important. I am in peer review induction so @JoeAldinger will finish this review

File: modules/persistent-storage-csi-drivers-supported.adoc [hunk 2]
```diff
@@ -67,14 +77,10 @@ ifndef::openshift-dedicated,openshift-rosa[]
 3.
 
 * Does not support offline snapshots or resize. Volume must be attached to a running pod.
+
+4.
```
Comment (JoeAldinger) [inline modules/persistent-storage-csi-drivers-supported.adoc:81]: The numbering doesn't look correct here in the preview. I _think_ you need `^1^` etc below the table for it to render correctly.

File: modules/persistent-storage-csi-drivers-supported.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoeAldinger) [reply modules/persistent-storage-csi-drivers-supported.adoc:34]: I'm not really sure what is going on here. If the callout and the paragraph are suppose to be before the table then we need `following`. If this move was unintentional then `preceding` is fine, but we need to put things back. Not sure which is correct.

**Review by JoeAldinger — COMMENTED** [review]
question and suggestion

**Comment by RomanBednar** [comment]
@lpettyjo lgtm

File: modules/persistent-storage-csi-drivers-supported.adoc [hunk 3]
```diff
@@ -67,14 +77,10 @@ ifndef::openshift-dedicated,openshift-rosa[]
...
+
+:FeatureName: Azure File cloning
```
Comment (gcharot) [inline modules/persistent-storage-csi-drivers-supported.adoc:83]: Azure File CSI Cloning

File: modules/persistent-storage-csi-drivers-supported.adoc [hunk 4]
```diff
@@ -24,6 +24,16 @@ supported by {product-title}
...
 and which CSI features they support, such as volume snapshots and resize.
 
+ifndef::openshift-rosa[]
+[IMPORTANT]
+====
+If your CSI driver is not listed in the following table, you must follow the installation instructions provided by your CSI storage vendor to use their supported CSI features.
+====
+endif::openshift-rosa[]
+ifdef::openshift-rosa[]
+In addition to the drivers listed in the following table, ROSA functions with CSI drivers from third-party storage vendors. Red Hat does not oversee third-party provisioners or the connected CSI drivers and the vendors fully control source code, deployment, operation, and Kubernetes compatibility. These volume provisioners are considered customer-managed and the respective vendors are responsible for providing support. See the link:https://docs.openshift.com/rosa/rosa_architecture/rosa_policy_service_definition/rosa-policy-responsibility-matrix.html#rosa-policy-responsibilities_rosa-policy-responsibility-matrix[Shared responsibilities for {product-title}] matrix for more information.
```
Comment (gcharot) [inline modules/persistent-storage-csi-drivers-supported.adoc:34]: Not sure who asked for this (relevant though) but shouldn't it also apply to ARO?

**Comment by gcharot** [comment]
Minor comments otherwise LGTM

File: modules/persistent-storage-csi-drivers-supported.adoc (hunk 4 above)
Comment (ropatil010) [reply modules/persistent-storage-csi-drivers-supported.adoc:34]: yeah agree, it worked fine on ARO, HCP Azure cluster as well. 
https://issues.redhat.com/browse/STOR-1719?focusedId=24599828&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-24599828

**Comment by ropatil010** [comment]
LGTM otherwise.!

File: modules/persistent-storage-csi-drivers-supported.adoc [hunk 5]
```diff
@@ -67,14 +77,14 @@ ifndef::openshift-dedicated,openshift-rosa[]
 3.
 
 * Does not support offline snapshots or resize. Volume must be attached to a running pod.
+
+4.
+
+* Azure File cloning does not supports NFS protocol. It supports the default storage class, which uses SMB protocol.
```
Comment (ropatil010) [inline modules/persistent-storage-csi-drivers-supported.adoc:83]: Hi @lpettyjo 
Sorry for the wrong wording comments [here](https://github.com/openshift/openshift-docs/pull/75937#discussion_r1604346731). Can we plz correct like this? 
It supports the default storage class, which uses SMB protocol => It supports with azurefile-csi storage class which uses SMB protocol. 

FI: Default sc will be managed-csi which already supported cloning feature long back.

File: modules/persistent-storage-csi-drivers-supported.adoc (hunk 3 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-drivers-supported.adoc:87]: Fixed.

File: modules/persistent-storage-csi-drivers-supported.adoc (hunk 4 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-drivers-supported.adoc:34]: @bmcelvee I believe that this is your content. OK with adding ARO to ifdef?

File: modules/persistent-storage-csi-drivers-supported.adoc (hunk 5 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-drivers-supported.adoc:83]: Fixed.

File: modules/persistent-storage-csi-drivers-supported.adoc (hunk 4 above)
Comment (bmcelvee) [reply modules/persistent-storage-csi-drivers-supported.adoc:34]: Thank you for bumping this @lpettyjo! It makes sense that it applies to ARO, but we don't have that content at this point in time. Looping in @bhardesty too.

Comment (bhardesty) [reply modules/persistent-storage-csi-drivers-supported.adoc:34]: @lpettyjo Sure, go ahead and add it! We don't currently publish any docs for ARO, but that will likely change in the future for ARO HCP.

Comment (lpettyjo) [reply modules/persistent-storage-csi-drivers-supported.adoc:34]: Done.

**Comment by gcharot** [comment]
LGTM

**Pull request merged**